File: backend/common/src/main/java/afsdigital/grahamselect/common/portfolio/domain/entities/Trade.java
package afsdigital.grahamselect.common.portfolio.domain.entities;

import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
public class Trade {
    private UUID id;
    private String userId;
    private String ticker;
    private String side; // COMPRA, VENDA
    private LocalDate tradeDate;
    private BigDecimal quantity;
    private BigDecimal price;
    private String broker;
    private LocalDateTime createdAt;
}

File: frontend/lib/src/features/portfolio/domain/entities/trade.dart
class Trade {
  final String id;
  final String userId;
  final String ticker;
  final String side;
  final DateTime tradeDate;
  final double quantity;
  final double price;
  final String broker;
  final DateTime createdAt;

  Trade({
    required this.id,
    required this.userId,
    required this.ticker,
    required this.side,
    required this.tradeDate,
    required this.quantity,
    required this.price,
    required this.broker,
    required this.createdAt,
  });
}
